javalib/src/main/scala/java/io/File.scala

```diff
@@ -264,7 +264,7 @@ class File(_path: String) extends Serializable with Comparable[File] {
 
   def list(filter: FilenameFilter): Array[String] =
     if (!isDirectory() || !canRead()) {
-      null
+      Array.empty[String]
```

Comment:
It was a crash with access to null pointer, there is no check for null later in the code